test/features/extensions/extensions.go

```diff
@@ -0,0 +1,76 @@
+//go:build e2e
+
+package extensions
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/features/cloudnative"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers"
+	componentDynakube "github.com/Dynatrace/dynatrace-operator/test/helpers/components/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/kubeobjects/secret"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/kubeobjects/statefulset"
+	"github.com/Dynatrace/dynatrace-operator/test/helpers/tenant"
+	"github.com/Dynatrace/dynatrace-operator/test/project"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	agSecretName                    = "ag-ca"
+	agCertificate                   = "custom-cas/agcrt.pem"
+	agCertificateAndPrivateKey      = "custom-cas/agcrtkey.p12"
+	agCertificateAndPrivateKeyField = "server.p12"
+	customPullSecretName            = "azurecr"
+	eecImageRepo                    = "extk8sregistry.azurecr.io/eec/dynatrace-eec"
+	eecImageTag                     = "1.302.0.20240916-161445"
+)
+
+func Feature(t *testing.T) features.Feature {
+	builder := features.New("extensions-components-rollout")
+
+	secretConfig := tenant.GetSingleTenantSecret(t)
+
+	options := []componentDynakube.Option{
+		componentDynakube.WithApiUrl(secretConfig.ApiUrl),
+		componentDynakube.WithCloudNativeSpec(cloudnative.DefaultCloudNativeSpec()),
```

Comment:
possible improvement:

cloudnative and oneagent stuff is not relevant for the extension's e2e test, I would not have it enabled, it would make the tests faster.

but be aware removing it might break: (hopefully it can handle crashloops)
```
builder.Assess("extensions execution controller started", statefulset.WaitFor(dynakube.ExtensionsExecutionControllerStatefulsetName, testDynakube.Namespace))
builder.Assess("extension collector started", statefulset.WaitFor(dynakube.ExtensionsCollectorStatefulsetName, testDynakube.Namespace))
```
because currently these have 0 effect, because the oneagent takes longer to get ready (around 2x time), so by the time you check the statefulsets they are already ready

Reply:
good catch, let me try to remove it.

Reply:
otherwise
```
=== RUN   TestStandard/extensions-components-rollout/extension_collector_started
--- PASS: TestStandard (174.75s)
    --- PASS: TestStandard/extensions-components-rollout (172.62s)
        --- PASS: TestStandard/extensions-components-rollout/create_custom_pull_secret (0.25s)
        --- PASS: TestStandard/extensions-components-rollout/create_AG_TLS_secret (0.13s)
        --- PASS: TestStandard/extensions-components-rollout/created_tenant_secret (0.27s)
        --- PASS: TestStandard/extensions-components-rollout/'dynakube'_dynakube_created (0.31s)
        --- PASS: TestStandard/extensions-components-rollout/oneagent_started (145.12s)
        --- PASS: TestStandard/extensions-components-rollout/'dynakube'_dynakube_phase_changes_to_'Running' (5.14s)
        --- PASS: TestStandard/extensions-components-rollout/extensions_execution_controller_started (5.16s)
        --- PASS: TestStandard/extensions-components-rollout/extension_collector_started (5.14s)
PASS
```

Reply:
fixed in e92229d3